### PR TITLE
[improvement](meta) Switch meta serialization to gson 3

### PR DESCRIPTION
**Pull request by iszhangpch**
## Proposed changes

Issue Number:

Switch meta serialization to gson
Contains the following classes：

Database
SmallFileMgr$SmallFile
catalog.Table
~~loadv2.LoadJob~~
loadv2.LoadJobFinalOperation

### CODE REVIEW

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

**Comment by iszhangpch** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java [hunk 1]
```diff
@@ -124,6 +125,7 @@ public enum DbState {
     @SerializedName(value = "dbProperties")
     private DatabaseProperty dbProperties = new DatabaseProperty();
 
+    @SerializedName(value = "bc")
     private BinlogConfig binlogConfig = new BinlogConfig();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java:129]: Do not need to persist.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java [hunk 2]
```diff
@@ -108,6 +108,7 @@ public class Database extends MetaObject implements Writable, DatabaseIf<Table>
     @SerializedName(value = "replicaQuotaSize")
     private volatile long replicaQuotaSize;
 
+    @SerializedName(value = "tq")
     private volatile long transactionQuotaSize;
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java:112]: Do not need to persist.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Table.java [hunk 3]
```diff
@@ -441,59 +442,61 @@ public TTableDescriptor toThrift() {
...
+                case VIEW:
+                case BROKER:
+                case ELASTICSEARCH:
+                case HIVE:
+                case JDBC:
+                    return GsonUtils.GSON.fromJson(json, Table.class);
+                default:
+                    throw new IOException("Unknown table type: " + type.name());
+            }
         }
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Table.java:491]: need recover nameToColumn.

Comment (dataroaring) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Table.java:491]: hasCompoundKey too.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/BrokerTable.java [hunk 4]
```diff
@@ -209,24 +214,7 @@ public TTableDescriptor toThrift() {
...
-        Text.writeString(out, lineDelimiter);
-        out.writeInt(brokerProperties.size());
-        for (Map.Entry<String, String> prop : brokerProperties.entrySet()) {
-            Text.writeString(out, prop.getKey());
-            Text.writeString(out, prop.getValue());
-        }
-    }
-
+    @Deprecated
     public void readFields(DataInput in) throws IOException {
```
Comment (SWJTU-ZhangLei) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/BrokerTable.java:218]: brokerTable and other external table need modify readField?

**Pull request closed without merging**